plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/ProgramConverter.kt

```diff
@@ -113,38 +114,46 @@ class ProgramConverter(val session: FirSession, override val config: PluginConfi
             UserFunctionEmbedding(callable)
         }
 
+    /**
+     * Returns an embedding of the class type, with details set.
+     */
     private fun embedClass(symbol: FirRegularClassSymbol): ClassTypeEmbedding {
         val className = symbol.classId.embedName()
-        return when (val existingEmbedding = classes[className]) {
-            null -> {
-                // The full class embedding is necessary to process the signatures of the properties of the class, since
-                // these take the class as a parameter. We thus do this in three phases:
-                // 1. Provide a class embedding in the `classes` map (necessary for embedType to not call this recursively).
-                // 2. Initialise the supertypes (including running this whole four-step process on each)
-                // 3. Initialise the fields
-                // 4. Process the properties of the class.
-                //
-                // With respect to the embedding, each phase is pure by itself, and only updates the class embedding at the end.
-                // This ensures the code never sees half-built supertype or field data. The phases can, however, modify the
-                // `ProgramConverter`.
-
-                // Phase 1
-                val newEmbedding = ClassTypeEmbedding(className, symbol.classKind == ClassKind.INTERFACE)
-                classes[className] = newEmbedding
-
-                // Phase 2
-                newEmbedding.initSuperTypes(symbol.resolvedSuperTypes.map(::embedType))
-
-                // Phase 3
-                val properties = symbol.propertySymbols
-                newEmbedding.initFields(properties.mapNotNull { processBackingField(it, symbol) }.toMap())
-
-                // Phase 4
-                properties.forEach { processProperty(it, newEmbedding) }
-                newEmbedding
-            }
-            else -> existingEmbedding
+        val embedding = classes.getOrPut(className) {
+            ClassTypeEmbedding(className, symbol.classKind == ClassKind.INTERFACE)
         }
```

Comment:
Intuitively `isInterface` should be included in `details`. Did you leave it as is because the situation is much simpler in this case?